### PR TITLE
ci: include image scanning as part of CI & use distroless/static:nonroot as the base images

### LINKED ISSUE #794: ci: include image scanning as part of CI

### LINKED ISSUE #805: Add "User" field to aad-pod-identity images
Currently there is no USER= in the add-pod-identity image.   When CIS scans are ran against an image without a User field the image is flagged for a HIGH Compliance issue as this represents a compromise. 

In order to remediate a User filed must be added to this image.  Steps to follow.

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

**Reason for Change**:

- Continuously monitoring for any image vulnerabilities with [trivy](https://github.com/aquasecurity/trivy)
- enabled docker buildkit for faster image building process (5 minutes -> under 2 minutes for building mic, nmi, identityvalidator and demo images)
- added "User" field to aad-pod-identity images and run MIC & NMI as root when deploying.

Fixes #794.
FIxes #805.

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [x] ran `make precommit`

**Issue Fixed**:

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

**Notes for Reviewers**:

### CODE REVIEW

File: test/e2e/liveness_probe_test.go [hunk 1]
```diff
@@ -3,55 +3,19 @@
...
-			Expect(err).To(BeNil())
-			if micPod.Name == micLeader.Name {
-				By(fmt.Sprintf("Ensuring that %s's health probe is active", micPod.Name))
-				Expect(strings.EqualFold(stdout, "Active")).To(BeTrue())
-			} else {
-				By(fmt.Sprintf("Ensuring that %s's health probe is not active", micPod.Name))
-				Expect(strings.EqualFold(stdout, "Not Active")).To(BeTrue())
-			}
-		}
-
```
Comment (chewong) [inline test/e2e/liveness_probe_test.go:54]: there is no more `clean-install` in MIC due to change in the base image.

File: test/e2e/framework/identityvalidator/identityvalidator_helpers.go [hunk 2]
```diff
@@ -56,10 +56,10 @@ func Create(input CreateInput) *corev1.Pod {
 				{
 					Name:  "identity-validator",
 					Image: fmt.Sprintf("%s/identityvalidator:%s", input.Config.Registry, input.Config.IdentityValidatorVersion),
-					Command: []string{
-						"sleep",
-						"3600",
```
Comment (chewong) [inline test/e2e/framework/identityvalidator/identityvalidator_helpers.go:61]: Same here. Implemented `identityvalidator --sleep` to workaround, which allows the container to sleep forever.

**Review by jchauncey — APPROVED** [review]
You could change all of the $(REGISTRY_NAME) type parameter calls to ${REGISTRY_NAME} since you are doing this in a shell. But its not a big deal if you dont. Other than that it looks good

File: Dockerfile [hunk 3]
```diff
@@ -1,30 +1,27 @@
...
-FROM base AS mic
-COPY --from=build /go/src/github.com/Azure/aad-pod-identity/bin/aad-pod-identity/mic /bin/
+FROM $BASEIMAGE AS mic
+COPY --from=builder /go/src/github.com/Azure/aad-pod-identity/bin/aad-pod-identity/mic /bin/
 ENTRYPOINT ["mic"]
 
-FROM base AS demo
-COPY --from=build /go/src/github.com/Azure/aad-pod-identity/bin/aad-pod-identity/demo /bin/
+FROM $BASEIMAGE AS demo
+COPY --from=builder /go/src/github.com/Azure/aad-pod-identity/bin/aad-pod-identity/demo /bin/
```
Comment (aramase) [inline Dockerfile:22]: Can we update the demo and identityvalidator images also to run as non-root?

File: test/e2e/liveness_probe_test.go (hunk 1 above)
Comment (aramase) [reply test/e2e/liveness_probe_test.go:54]: Not a blocker for this PR, but we should explore doing the health check with podIP from the busybox container which runs on host network

File: Dockerfile (hunk 3 above)
Comment (aramase) [reply Dockerfile:22]: Resolving this as base image is `nonroot`

**Review by aramase — APPROVED** [review]
lgtm

File: Dockerfile (hunk 3 above)
Comment (chewong) [reply Dockerfile:22]: Both of them should be running as non-root since the base image is `gcr.io/distroless/static:nonroot`

**Pull request merged**